### PR TITLE
Add warning popup when loading save states for the first time

### PR DESCRIPTION
**Pull request by GPUCode**
Also removed some global state in GMainWindow because it bothered me and added the GCC 13/MSVC compile fix

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6565)

### CODE REVIEW

**Review by Steveice10 — APPROVED** [review]

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -1940,19 +1927,34 @@ void GMainWindow::TriggerRotateScreens() {
...
 
 void GMainWindow::OnLoadState() {
     QAction* action = qobject_cast<QAction*>(sender());
-    assert(action);
+    ASSERT(action);
 
-    Core::System::GetInstance().SendSignal(Core::System::Signal::Load, action->data().toUInt());
-    Core::System::GetInstance().frame_limiter.AdvanceFrame();
+    if (Settings::values.show_save_state_warning) {
+        QMessageBox msg(this);
```
Comment (FearlessTobi) [inline src/citra_qt/main.cpp:1942]: nit: you can construct a QMessageBox::warning directly, like we do in `GMainWindow::LoadAmiibo` for example

File: src/citra_qt/configuration/config.cpp [hunk 2]
```diff
@@ -1033,6 +1034,7 @@ void Config::SaveMiscellaneousValues() {
     qt_config->beginGroup(QStringLiteral("Miscellaneous"));
 
     WriteBasicSetting(Settings::values.log_filter);
+    WriteBasicSetting(Settings::values.show_save_state_warning);
```
Comment (SachinVin) [inline src/citra_qt/configuration/config.cpp:1037]: I think this should be in `UISettings`, UI only stuff generally seems to be placed there

**Comment by stuken** [comment]
On canary this is causing a nullptr error on closing after a game has loaded.
https://github.com/citra-emu/citra/blob/848116b5be6be6c7d24ac099a579e19fc21f2c27/src/citra_qt/bootmanager.cpp#L591
The focusOutEvent handler is getting called after Keyboard has been destroyed.

**Comment by GPUCode** [comment]
> On canary this is causing a nullptr error on closing after a game has loaded.
> 
> https://github.com/citra-emu/citra/blob/848116b5be6be6c7d24ac099a579e19fc21f2c27/src/citra_qt/bootmanager.cpp#L591
> 
> The focusOutEvent handler is getting called after Keyboard has been destroyed.

Is the crash related to this PR? I haven't touched that code

**Comment by stuken** [comment]
> Is the crash related to this PR? I haven't touched that code

Verified again on the latest canary. Crashes with the published and local builds. Revert this change and it shuts down cleanly.

**Comment by Steveice10** [comment]
This is the trace I get closing Citra while in a game on the latest PR artifact, using MSVC:
```
citra_qt!boost::archive::detail::pointer_oserializer<boost::archive::binary_oarchive,Kernel::AddressArbiter::Callback>::save_object_ptr+0x132c6
citra_qt!boost::serialization::singleton_module::unlock+0x77cc
Qt6Widgets!QWidget::event+0x52d
citra_qt!boost::serialization::singleton_module::unlock+0x75f9
Qt6Widgets!QApplicationPrivate::notify_helper+0x10e
Qt6Widgets!QApplication::notify+0x1891
Qt6Core!QCoreApplication::notifyInternal2+0xc5
Qt6Widgets!QApplicationPrivate::setFocusWidget+0x112
Qt6Widgets!QApplicationPrivate::setActiveWindow+0x784
Qt6Widgets!QWidgetPrivate::hide_sys+0xb0
Qt6Widgets!QWidgetPrivate::hide_helper+0xb8
Qt6Widgets!QWidgetPrivate::setVisible+0x315
Qt6Widgets!QWidgetPrivate::handleClose+0x101
Qt6Widgets!QWidgetRepaintManager::updateStaticContentsSize+0x6cc
Qt6Gui!QWindow::event+0x268
Qt6Widgets!QWidgetRepaintManager::updateStaticContentsSize+0x116a
Qt6Widgets!QApplicationPrivate::notify_helper+0x10e
Qt6Widgets!QApplication::notify+0x1891
Qt6Core!QCoreApplication::notifyInternal2+0xc5
Qt6Gui!QGuiApplicationPrivate::processCloseEvent+0xab
Qt6Gui!QWindowSystemInterface::sendWindowSystemEvents+0xe8
Qt6Core!QEventDispatcherWin32::processEvents+0x90
Qt6Gui!QWindowsGuiEventDispatcher::processEvents+0x19
Qt6Core!QEventLoop::exec+0x194
Qt6Core!QCoreApplication::exec+0x15d
citra_qt!boost::serialization::singleton<boost::serialization::extended_type_info_typeid<Service::AM::AM_NET> >::singleton<boost::serialization::extended_type_info_typeid<Service::AM::AM_NET> >+0x81bea
citra_qt!boost::serialization::singleton<boost::archive::detail::extra_detail::map<boost::archive::binary_oarchive> >::is_destroyed+0xec0fd
citra_qt!boost::serialization::singleton<boost::archive::detail::extra_detail::map<boost::archive::binary_oarchive> >::is_destroyed+0xeb1d2
kernel32!BaseThreadInitThunk+0x1d
ntdll!RtlUserThreadStart+0x28
```

**Comment by Steveice10** [comment]
Removing the added [gameList->setDisabled(true);](https://github.com/citra-emu/citra/pull/6565/commits/5344a815adf0014d635b42c05eecac9033023878#diff-34f48db7948cae490563670110ed0d9f6384d971531021230d9995475a44a440R1189) fixes the crash, so it may be a latent bug that this exposed.

**Comment by stuken** [comment]
On nightly you can also crash with a pretty much identical backtrace when you have the filter list disabled. Add nullptr check before InputCommon::GetKeyboard()->ReleaseAllKeys(); and call it a day I guess.

**Comment by GPUCode** [comment]
Global state being cringe moment

File: src/video_core/renderer_opengl/gl_texture_runtime.cpp [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2023 Citra Emulator Project
 // Licensed under GPLv2 or any later version
 // Refer to the license.txt file included.
-
```
Comment (FearlessTobi) [inline src/video_core/renderer_opengl/gl_texture_runtime.cpp:4]: This should not have been removed

**Review by FearlessTobi — COMMENTED** [review]
Can we maybe seperate the global state changes to a different PR?
I'd like to get the savestate warning and the gcc fix merged soon.

**Comment by GPUCode** [comment]
Fixed the errors, merging to unblock other PRs

**Pull request merged**